src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -456,15 +901,19 @@ public int OnItemChosen(string pszTitle, string pszPath)
 
                 var expansion = EditorAdaptersFactoryService.GetBufferAdapter(textViewModel.DataBuffer) as IVsExpansion;
                 earlyEndExpansionHappened = false;
-                hr = expansion.InsertNamedExpansion(pszTitle, pszPath, textSpan, this, LanguageServiceGuid, fShowDisambiguationUI: 0, pSession: out ExpansionSession);
+
+                // This expansion was chosen from the snippet picker, and not derived from a symbol. Make sure the state
+                // isn't tracking any symbol information.
+                _state.ClearSymbolInformation();
```

Comment:
Any reason this can't just be Clear()? Yeah, it'll clear the value you're overwriting in the next line, but would mean it's more impervious to any other refactoring where somebody puts some other non-symbol state in that matters.

Reply:
➡️ The expansion session is still active at this point. `Clear` would not accurately reflect the state.

Reply:
Which session? It's not starting until the next line, right? If there was a prior session it seems like we aren't caring about it at this point?

It's possible I'm just missing the scenario here -- I would have presumed this method is only called on the snippet picker UI and so at that point any session going away is moot, and there's nothing to preserve because it's not going to be a method expansion snippet being invoked.

Reply:
The presence of an active snippet session does not prevent the snippet picker from working, so while it's not common for this case to be hit with an active non-null session it is possible.